rdt-open-reader/src/main/java/io/ona/rdt/application/RDTApplication.java

```diff
@@ -153,4 +165,23 @@ public ParasiteProfileRepository getParasiteProfileRepository() {
         }
         return parasiteProfileRepository;
     }
+
+    public void verifyUserAuthorization() {
+
+        if (future != null) {
+            future.cancel(true);
+        }
+
+        future = executor.submit(() -> {
```

Comment:
For consistency, we should use an AsyncTask here

Reply:
AsyncTask was deprecated in API level 30

Reply:
It's still usable, I think if you want to change this you'd want to change all other places AsyncTask is used for consistency.

Reply:
That can be a separate issue

Reply:
done